api/src/room_store/s3_room_archive.py

```diff
@@ -0,0 +1,58 @@
+import json
+from dataclasses import asdict
+from typing import Iterable, AsyncIterator
+
+import botocore.exceptions
+from aiobotocore.client import AioBaseClient
+
+from src.api.api_structures import Action
+from src.room_store.json_to_actions import json_to_actions
+from src.room_store.room_archive import RoomArchive
+
+ROOM_DIR = 'rooms/'
+
+
+def _room_id_to_key(room_id: str) -> str:
+    return f'{ROOM_DIR}{room_id}'
+
+
+class S3RoomArchive(RoomArchive):
+    def __init__(self, client: AioBaseClient, bucket: str):
+        self._client = client
+        self._bucket = bucket
+
+    async def get_all_room_ids(self) -> AsyncIterator[str]:
+        paginator = self._client.get_paginator('list_objects')
+        page_iterator = paginator.paginate(Bucket=self._bucket)
+        async for page in page_iterator:
+            for item in page.get('Contents', []):
+                yield item['Key'].lstrip(ROOM_DIR)
+
+    async def room_exists(self, room_id: str) -> bool:
+        try:
+            await self._client.head_object(
+                Bucket=self._bucket, Key=_room_id_to_key(room_id)
+            )
+        except botocore.exceptions.ClientError:
+            return False
+        return True
+
+    async def read(self, room_id: str) -> Iterable[Action]:
+        resp = await self._client.get_object(
+            Bucket=self._bucket, Key=_room_id_to_key(room_id)
+        )
+        data = await resp['Body'].read()
+        print(data)
+        return json_to_actions([str(data.decode())])
+
+    async def write(self, room_id: str, data: Iterable[Action]) -> None:
```

Comment:
I think we need a test the verifies that writing to the same room twice works as we expect. (In case s3 defaults to append or throws an error when you write to a file that already has content)